src/services/static-snapshot-service.ts

```diff
@@ -1,20 +1,113 @@
+import * as bodyParser from 'body-parser'
+import * as cors from 'cors'
+import * as express from 'express'
+import {Server} from 'http'
+import * as puppeteer from 'puppeteer'
+import * as walk from 'walk'
 import logger from '../utils/logger'
+import {agentJsFilename} from '../utils/sdk-utils'
 import {StaticSnapshotOptions} from './static-snapshot-options'
 
+// Use this instead of importing PercyAgent - we only want the compiled version
+declare var PercyAgent: any
+
 export default class StaticSnapshotService {
+  private readonly app: express.Application
+  private readonly options: StaticSnapshotOptions
+  private server: Server | null = null
+
   constructor(options: StaticSnapshotOptions) {
-    // logger.info('calling constructor...')
+    this.app = express()
+    this.options = options
+
+    this.app.use(cors())
+    this.app.use(bodyParser.urlencoded({extended: true}))
+    this.app.use(bodyParser.json({limit: '50mb'}))
+
+    this.app.use(options.baseUrl, express.static(options.snapshotDirectory))
   }
 
   async start() {
-    // logger.info('starting static snapshot service...')
+    logger.info('starting static snapshot service...')
+    this.server = await this.app.listen(this.options.port)
   }
 
   async snapshotAll() {
-    // logger.info('taking snapshots of the static site...')
+    logger.info('taking snapshot of static site...')
+
+    const browser = await puppeteer.launch({
+      args: ['--no-sandbox'],
+      handleSIGINT : false,
+    })
+
+    const page = await browser.newPage()
+
+    const pageUrls = await this._buildPageUrls()
+
+    for (const url of pageUrls) {
+      await page.goto(url)
+      const percyAgentClientFilename = agentJsFilename()
+
+      await page.addScriptTag({
+        path: percyAgentClientFilename,
+      })
+
+      await page.evaluate((name) => {
+        const percyAgentClient = new PercyAgent()
+        return percyAgentClient.snapshot(name)
+      }, url)
+    }
+
+    browser.close()
   }
 
   async stop() {
-    // logger.info('stopping static snapshot service...')
+    logger.info('stopping static snapshot service...')
+
+    if (this.server) { await this.server.close() }
+  }
+
+  // provide a simple way to test that the constructor recieved the expected arguments
+  _getOptions() {
+    return this.options
+  }
+
+  async _buildPageUrls() {
+    const baseUrl = `http://localhost:${this.options.port}${this.options.baseUrl}`
+    const pageUrls = [] as any
+
+    const walkOptions = {
```

Comment:
Let's consider using something like https://www.npmjs.com/package/glob it will allow us to provide a pattern and get back a list of files. There's even an `ignore` option we could use.